node/src/components/deploy_buffer.rs

```diff
@@ -202,6 +217,9 @@ impl DeployBuffer {
         let mut holds = vec![];
         for (deploy_hash, footprint, approvals) in self.proposable() {
             let with_approvals = DeployHashWithApprovals::new(deploy_hash, approvals);
+            if !self.gossiped_deploy_hashes.remove(&deploy_hash) {
```

Comment:
I'd like to keep it as is, especially that the guarantee is in a different function. It may happen that someone removes the filter in `fn proposable()` without updating `fn appendable_block()`